### PR TITLE
[DNM] Add multi-user support for openstack

### PR DESCRIPTION
**Pull request by kshtsk**
Make it possible to run simultaneously two and more teuthology server setups in the same openstack region.

### CODE REVIEW

File: teuthology/openstack/__init__.py [hunk 1]
```diff
@@ -923,40 +933,43 @@ def get_user_data(self):
         log.debug("get_user_data: " + content + " written to " + path)
         return path
 
+    def server_group(self):
+        return "%s" % self.args.name
+
+    def worker_group(self):
+        return  "%s-worker" % self.args.name
+
```
Comment (smithfarm) [inline teuthology/openstack/__init__.py]: Can we add a "teuthology-" namespace prefix to the `server_group` and `worker_group` values, please? 

The reason why this is preferable: without a namespace, a user might run t-o with `--name foo` in an environment with an existing `foo` security group that is unrelated to teuthology.

Comment (smithfarm) [reply teuthology/openstack/__init__.py]: The same applies to the teuthology and packages-repository VMs themselves - if `--name foo` is given, then the teuthology VM should not be called "foo", because there might already be a VM with that name. Of course, openstack will still create the new VM "foo" but it's very confusing for users looking at the list of instances in the region when they see two instances both called "foo".

**Comment by smithfarm** [comment]
**testing**

- [x] single test run without `--name` to create cluster "teuthology" (passed)
- [ ] second test run with `--name foo` to create cluster "foo"
- [ ] second test run without `--name` targeting cluster "teuthology"
- [ ] third test run with `--name foo` targeting cluster "foo"

**Comment by smithfarm** [comment]
@kshtsk This is `--name foo` http://167.114.234.197:8081/

Notice the first run succeeded - this included building the RPMs, of course. The second two (failed) runs were both started while the first one was still building RPMs. The failure "HTTPConnectionPool(host='2001', port=80): Max retries exceeded with url: /ceph-rpm-sle12-x86_64-basic/sha1/232120f8d95a0765c6536fae1f5e681de9cc0689/version (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x7fce7224b490>: Failed to establish a new connection: [Errno 22] Invalid argument',))" is not one I've ever seen before.

**Comment by smithfarm** [comment]
`host='2001'` would indicate that it's mistaking the IPv6 address for the hostname, maybe?

**Comment by smithfarm** [comment]
```
2017-10-10T14:04:02.796 DEBUG:teuthology.misc:System to be installed: SUSE
2017-10-10T14:04:02.797 INFO:teuthology.task.install.rpm:Pulling from http://2001/ceph-rpm-sle12-x86_64-basic/sha1/232120f8d95a0765c6536fae1f5e681de9cc0689
2017-10-10T14:04:02.797 INFO:teuthology.packaging:Looking for package version: http://2001/ceph-rpm-sle12-x86_64-basic/sha1/232120f8d95a0765c6536fae1f5e681de9cc0689/version
```

Ah, looks like the PR broke something in the install task.

**Comment by kshtsk** [comment]
@smithfarm I guess there is a name conflict for builder server which starts from 'ceph-rpm' we need to prefix it with --name.

File: teuthology/openstack/__init__.py (hunk 1 above)
Comment (kshtsk) [reply teuthology/openstack/__init__.py]: If we add everywhere prefix like 'teuth-' this will not allow to teardown existing systems. Also, you will not be possible to give server name exactly which you like because it will always have 'teuth-' prefix. As a convention one can always give `--name=teuthology-foo` instead `--name=foo`.

Comment (smithfarm) [reply teuthology/openstack/__init__.py]: > If we add everywhere prefix like teuth- this will not allow to teardown existing systems.

By this, you mean systems running from before this PR is merged? I don't think that's particularly important - those can be deleted manually.

Comment (smithfarm) [reply teuthology/openstack/__init__.py]: > you will not be possible to give server name exactly which you like because it will always have teuth- prefix.

Always having the prefix is the whole point. We cannot rely on users to enter the prefix by themselves.

Keep in mind that the tenant/region may be used by multiple people and not all of them are necessarily using teuthology.

With the current master, it is always clear which VMs belong to the teuthology environment. But if we merge this PR as it is, this will no longer be the case. We will suddenly start having teuthology VMs named "foo", "bar", etc. This would be a regression because it would no longer be possible, simply by inspecting the name, to tell what the VM is doing.

**Comment by kshtsk** [comment]
@smithfarm I pushed new commit, you can try and rerun tests.

**Comment by kshtsk** [comment]
Created another PR #105

**Comment by kshtsk** [comment]
This ticket is replaced with #105 that's why I'm closing it.
